paparazzi-gradle-plugin/src/main/java/app/cash/paparazzi/gradle/PaparazziPlugin.kt

```diff
@@ -180,26 +182,24 @@ class PaparazziPlugin : Plugin<Project> {
         task.paparazziResources.set(buildDirectory.file("intermediates/paparazzi/${variant.name}/resources.txt"))
       }
 
-      val testVariantSlug = testVariant.name.capitalize(Locale.US)
-
       project.plugins.withType(JavaBasePlugin::class.java) {
-        project.tasks.named("compile${testVariantSlug}JavaWithJavac")
-          .configure { it.dependsOn(writeResourcesTask) }
+        testVariant.javaCompileProvider.configure { it.dependsOn(writeResourcesTask) }
       }
 
-      project.plugins.withType(KotlinMultiplatformPluginWrapper::class.java) {
-        val multiplatformExtension =
-          project.extensions.getByType(KotlinMultiplatformExtension::class.java)
-        check(multiplatformExtension.targets.any { target -> target is KotlinAndroidTarget }) {
+      if (project.plugins.hasPlugin(KotlinMultiplatformPluginWrapper::class.java)) {
```

Comment:
Will this create a plugin ordering requirement between Paparazzi and KGP? Or is it handled somehow?

Reply:
Yes. Should still use a with function to configure as before. If you need to detect to case of no Kotlin plugin you can set a var to true when a plugin is detected and check it was set to true in afterEvaluate

Reply:
Similar to this: https://github.com/cashapp/paparazzi/pull/1111#issuecomment-1739779868

I'm not quite satisfied with the execution flow here, given how the snapshotSourceSet is either determined via async callbacks (making it challenging to have all the logic in the same place) or sync hasPlugin calls (introducing a plugin ordering problem).  Will probably need to refactor the plugin code more to make this easier.